### ISSUE TITLE
Is checking token.Valid needed?

### ISSUE THREAD

**Issue by colegleason** [dgrijalva/jwt-go#33]
In the README this example is given:
```
    token, err := jwt.Parse(myToken, func(token *jwt.Token) ([]byte, error) {
        return myLookupKey(token.Header["kid"])
    })

    if err == nil && token.Valid {
        deliverGoodness("!")
    } else {
        deliverUtterRejection(":(")
    }
```

Is there any case where `err == nil` but `token.Valid == false` or can we just check `err`?

**Comment by dgrijalva** [comment]
Fix for #75

**Comment by dgrijalva** [comment]
Added tests to ensure consistency: c8d4fff6cc42e4e5dd48ffcc95e63446432180e8
